### PR TITLE
Addition of a Stokes flow example (steady and unsteady)

### PR DESCRIPTION
**Pull request by lokmanb**
Draft PR for the addition of a Stokes flow example. The example uses a mixed formulation and Taylor-Hood elements for the discretization. The steady case is based on a lid driven triangular cavity which exhibits Moffat vortices. The unsteady case is an oscillating flat plate flow for which an analytical solution (for a semi-infinite domain) is available.

### CODE REVIEW

File: src/example/stokes_flow/stokes_flow.jl [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+        mesh,
+        U_vel,
+    )
+    vars_2 = Dict("Pressure" => pressure)
+    Bcube.write_vtk_lagrange(
+        joinpath(outputpath, "output_pressure_steady"),
+        vars_2,
+        mesh,
+        U_pre,
+    )
```
Comment (ghislainb) [inline src/example/stokes_flow/stokes_flow.jl:163]: could you add a comment to explain why it's better to write results in two seperate vtk files when solutions have different degrees?

File: src/example/stokes_flow/stokes_flow.jl [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+    # Solve linear system
+    sol = A \ L
+
+    sol = sol .+ Wd
+
+    ϕ = FEFunction(V)
+
+    # Write solution
+    set_dof_values!(ϕ, sol)
```
Comment (ghislainb) [inline src/example/stokes_flow/stokes_flow.jl:147]: `ϕ = FEFunction(V, sol)` ?

File: src/example/stokes_flow/stokes_flow.jl [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        xc = 0.5e-1,
+        yc = 2.5e-1,
+    )
+    mesh = read_msh(mesh_path)
+
+    fsu = FunctionSpace(fspace, degree_u)
+    U_vel = TrialFESpace(
+        fsu,
+        mesh,
+        Dict("South" => (x, t) -> [sin(2.0 * π * f * t), 0.0]);
```
Comment (ghislainb) [inline src/example/stokes_flow/stokes_flow.jl:250]: use static array : `SA[sin...]`

File: src/example/stokes_flow/stokes_flow.jl [hunk 4]
```diff
@@ -0,0 +1,354 @@
+module stokes_flow_API #hide
```
Comment (ghislainb) [inline src/example/stokes_flow/stokes_flow.jl:1]: I suggest the following structure:
# Stokes flow example (FE with Taylor-Hood elements)
## Steady case : Moffat vortices
### Theory 
### ...
## Unsteady case : oscillating plate
### Theory 
### ...

We could discuss it.

File: src/example/stokes_flow/stokes_flow.jl (hunk 1 above)
Comment (lokmanb) [reply src/example/stokes_flow/stokes_flow.jl:163]: In fact different degrees weren't necessary for the visualization. I will change this to have a single output file.

File: src/example/stokes_flow/stokes_flow.jl [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+    fsp = FunctionSpace(fspace, degree_p)
+    U_pre = TrialFESpace(fsp, mesh, Dict("edge_top" => SA[0.0]); size = 1)
+    V_pre = TestFESpace(U_pre)
+
+    # Define MultiFESpace (mixed formalism)
+    U = MultiFESpace(U_vel, U_pre)
+    V = MultiFESpace(V_vel, V_pre)
+
+    velocity = FEFunction(U_vel, 0.0)
```
Comment (bmxam) [inline src/example/stokes_flow/stokes_flow.jl:113]: `velocity = FEFunction(U_vel)` initializes with zeros. But if you prefer a more explicit init, you can leave it as is.

File: src/example/stokes_flow/stokes_flow.jl [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+        erfc(sqrt(0.5 * T) * (C + Y / (T * sqrt(2)))),
+    )
+    return Uₛ + Uₜ
+end
+
+# Function that runs the unsteady case:
+function run_unsteady()
+    # Read 2D mesh
+    mesh_path = joinpath(outputpath, "mesh.msh")
+    gen_rectangle_mesh(
```
Comment (bmxam) [inline src/example/stokes_flow/stokes_flow.jl:223]: If you use a quad mesh, why not using `rectangle_mesh` from Bcube? If the idea is to easily allow a switch to triangles, it's ok.

File: src/example/stokes_flow/stokes_flow.jl [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+    V_vel = TestFESpace(U_vel)
+
+    fsp = FunctionSpace(fspace, degree_p)
+    U_pre = TrialFESpace(fsp, mesh, Dict("South" => SA[0.0]); size = 1)
+    V_pre = TestFESpace(U_pre)
+
+    U = MultiFESpace(U_vel, U_pre)
+    V = MultiFESpace(V_vel, V_pre)
+
+    velocity = FEFunction(U_vel, 0.0)
```
Comment (bmxam) [inline src/example/stokes_flow/stokes_flow.jl:251]: same remark

File: src/example/stokes_flow/stokes_flow.jl [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+    # definition of bilinear and linear forms
+    m((u, p), (v, q)) = ∫(ρ * u ⋅ v)dΩ
+    a((u, p), (v, q)) = ∫(μ * ∇(u) ⊡ ∇(v) - tr(∇(v)) * p + tr(∇(u)) * q)dΩ
+    l((v, q)) = ∫(fv ⋅ v)dΩ
+
+    M = assemble_bilinear(m, U, V)
+    A = assemble_bilinear(a, U, V)
+    L = assemble_linear(l, V)
+
+    M0 = copy(M)
```
Comment (bmxam) [inline src/example/stokes_flow/stokes_flow.jl:270]: where do you use it?

**Review by bmxam — APPROVED** [review]
very nice !

**Pull request merged**